### PR TITLE
Improve dependent beans calculation

### PR DESCRIPTION
**Pull request by EgorkaKulikov**
## Description

Running Spring code concretely requires resetting some beans because they affect Spring context.
This PR improves the list of beans to reset.

Will be replaced with a better designed solution later, TODO is left in the code.

## How to test

Standalone testing is not required, just as a part of full Spring integration test generation testing.

## Self-check list

- [x] I've set the proper **labels** for my PR (at least, for category and component).
- [x] PR **title** and **description** are clear and intelligible.
- [x] I've added enough **comments** to my code, particularly in hard-to-understand areas.
- [ ] The functionality I've repaired, changed or added is covered with **automated tests**.
- [ ] **Manual tests** have been provided optionally.
- [x] The **documentation** for the functionality I've been working on is up-to-date.

### CODE REVIEW

File: utbot-core/src/main/kotlin/org/utbot/common/ClassLoaderUtil.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.utbot.common
+
+import java.net.URLClassLoader
+
+/**
+ * Checks that the class given by its binary name can be loaded with given classLoader.
+ */
+fun URLClassLoader.canLoad(classBinaryName: String): Boolean {
```
Comment (IlyaMuravjov) [inline utbot-core/src/main/kotlin/org/utbot/common/ClassLoaderUtil.kt:8]:
```suggestion
fun URLClassLoader.isOnClasspath(classBinaryName: String): Boolean {
```
Also, the current version of documentation fails to convey that this function will return `true` despite `loadClass` throwing `NoClassDefFoundError` when the class is on class path, while it's parent or some interfaces are not.

File: utbot-framework/src/main/kotlin/org/utbot/sarif/SarifReport.kt [hunk 2]
```diff
@@ -360,14 +360,14 @@ class SarifReport(
...
-            instruction.className to instruction.methodSignature // group by method
+            instruction.classFqn to instruction.methodSignature // group by method
         }.map { (_, instructionsForOneMethod) ->
             instructionsForOneMethod.last() // we need only last to construct the stack trace
         }
 
         val sarifExecutionTrace = executionTrace.map { instruction ->
             resolveStackTraceElementByNames(
-                classFqn = instruction.className.replace('/', '.'),
+                classFqn = instruction.classFqn.replace('/', '.'),
```
Comment (IlyaMuravjov) [inline utbot-framework/src/main/kotlin/org/utbot/sarif/SarifReport.kt:370]: `instruction.classFqn.replace('/', '.')` seems to repeat multiple times throughout the code base. Consider adding `val className: String get() = classFqn.replace('/', '.')` in `Instruction` class itself.

File: utbot-core/src/main/kotlin/org/utbot/common/ClassLoaderUtil.kt (hunk 1 above)
Comment (EgorkaKulikov) [reply utbot-core/src/main/kotlin/org/utbot/common/ClassLoaderUtil.kt:8]: hasOnClasspath

File: utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/CoverageApi.kt [hunk 3]
```diff
@@ -3,15 +3,15 @@ package org.utbot.framework.plugin.api
 /**
  * Represents a covered bytecode instruction.
  *
- * @param className the fqn in internal form, i.e. com/rest/order/services/OrderService$InnerClass.
+ * @param classFqn the fqn in internal form, i.e. com/rest/order/services/OrderService$InnerClass.
```
Comment (IlyaMuravjov) [inline utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/CoverageApi.kt:6]: Consider naming this property `internalName` so it's more clear that it has slashes

**Review by IlyaMuravjov — APPROVED** [review]

**Pull request merged**